src/proxy/libp2p_proxy_server.erl

```diff
@@ -99,8 +99,23 @@ handle_call({init_proxy, ID, ServerStream, SAddress}, _From, #state{data=Data,
     Data1 = maps:put(ID, PState, Data),
     self() ! {post_init, ID, SAddress},
     {reply, ok, State#state{data=Data1}};
-handle_call({init_proxy, ID, _ServerStream, SAddress}, _From, State) ->
+handle_call({init_proxy, ID, _ServerStream, SAddress}, _From, #state{swarm=Swarm}=State) ->
     lager:warning("cannot process proxy request for ~p, ~p. Limit exceeded", [ID, SAddress]),
+    erlang:spawn(
+        fun() ->
+            case libp2p_proxy:dial_framed_stream(Swarm, SAddress, [{id, ID}, {swarm, Swarm}]) of
+                {ok, ClientStream} ->
+                    Ref = erlang:monitor(process, ClientStream),
+                    ClientStream ! proxy_overloaded,
+                    receive
+                        {'DOWN', Ref, process, ClientStream, _} ->
```

Comment:
Should we have a timeout here, maybe one that would kill the clientstream process?